### PR TITLE
Updated action.js to Handle Locked LeetCode Premium Problems and Improve Sync Reliability

### PR DESCRIPTION
**Pull request by AlyHG**
**Description:**
This pull request addresses an issue where users with previously unlocked LeetCode premium problems would experience failures during the sync process. The following changes have been made:

- **Error Handling for Locked Problems:** Added error handling in the `getInfo` function to skip locked problems. When the function encounters a locked problem (HTTP 403 error), it logs a message and skips the problem instead of retrying or throwing an exception.
- **Fetching Question Data:** Updated the `getQuestionData` function to handle errors when fetching question data for locked problems. If fetching the question data results in a locked problem error (HTTP 403), it logs the error and returns null to indicate that the problem should be skipped.
- **Sync Function Modification:** Modified the `sync` function to continue syncing other problems even if some are locked, ensuring that locked problems do not cause the entire sync process to fail.
- **Additional Logging:** Added logging to help identify and skip locked problems during the sync process.

These changes improve the sync process by handling scenarios where some LeetCode problems are locked, allowing the action to continue syncing all available, unlocked problems successfully.

### CODE REVIEW

File: src/action.js [hunk 1]
```diff
@@ -415,13 +425,22 @@ async function sync(inputs) {
       leetcodeSession,
       leetcodeCSRFToken,
     );
+    if (!submission) {
```
Comment (joshcai) [inline src/action.js:428]: Could you also do "submission === null" here to be consistent with L440?

File: README.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ GitHub Action for automatically syncing LeetCode submissions to a GitHub reposit
 
        steps:
          - name: Sync
-           uses: joshcai/leetcode-sync@v1.6
+           uses: AlyHG/leetcode-sync@master
```
Comment (joshcai) [inline README.md:51]: Could you revert this line? I'll create a new release after this is in, and bump the version number up.

File: README.md [hunk 3]
```diff
@@ -111,3 +111,17 @@ Special thanks to the following people who helped beta test this GitHub Action a
 - [uakfdotb](https://github.com/uakfdotb)
 - [hexecute](https://github.com/hexecute)
 - [JonathanZhu11](https://github.com/JonathanZhu11)
+
+## Updated by [Aly Ghallab](https://github.com/AlyHG) to Handle Locked LeetCode Premium Problems and Improve Sync Reliability - June 19, 2024
```
Comment (joshcai) [inline README.md:115]: Would it be okay if I add this info in the release notes section when creating a new release? e.g. similar to https://github.com/joshcai/leetcode-sync/releases/tag/v1.6

**Review by joshcai — COMMENTED** [review]
Thanks for the improvement, really appreciate it!

**Comment by AlyHG** [comment]
Hey, I applied your feedback and made the changes requested. Let me know if there's anything else.

**Pull request merged**